### PR TITLE
Make @property deprecatable

### PR DESCRIPTION
**Pull request by indrekj**
screen shot 2016-09-14 at 13 52 57

### CODE REVIEW

File: templates/class.hamlc [hunk 1]
```diff
@@ -79,7 +79,7 @@
           - if @entity.properties.length > 0
             %dl.properties
               - for property in @entity.properties
-                %dt{id: @anchorFor(property)}
+                %dt{id: @anchorFor(property), deprecated: if property.documentation?.deprecated then true else false}
```
Comment (deiwin) [inline templates/class.hamlc]: isn't this equal to `deprecated: property.documentation?.deprecated?`?

Comment (deiwin) [inline templates/class.hamlc]: Okay, not equal if `deprecated` is a boolean, but that can't happen anyway, right?

File: templates/partials/method_summary.hamlc [hunk 2]
```diff
@@ -21,9 +21,11 @@
       - if method.documentation?.deprecated
         %span.deprecated.note.title Deprecated
 
-      %span.desc
-        - if method.documentation?.deprecated
+      - if method.documentation?.summary
+        %span.desc
+          != @activate method.documentation?.summary, @path, true
```
Comment (deiwin) [inline templates/partials/method_summary.hamlc]: The `?.` is redundant now. Also, why did you need to change this?

Comment (deiwin) [inline templates/partials/method_summary.hamlc]: Same redundant `?.` below

File: templates/class.hamlc (hunk 1 above)
Comment (indrekj) [inline templates/class.hamlc]: I copied this from https://github.com/salemove/codo-theme-sm/blob/deprecated-property/templates/partials/sidebar_method_list.hamlc#L3 . I could probably make it shorter but then I think I should change both. What do you think? Do we want to keep our code as similar to the existing code base or not?

File: templates/partials/method_summary.hamlc (hunk 2 above)
Comment (indrekj) [inline templates/partials/method_summary.hamlc]: I made it consistent with the `property` docs. Everything else were using `.note.deprecated` class expect for this one.

File: templates/class.hamlc (hunk 1 above)
Comment (deiwin) [inline templates/class.hamlc]: We've changed enough from the default theme that pulling updates from there back into our theme is quite difficult anyway. So no real reason to avoid updating, imo.

File: templates/partials/method_summary.hamlc (hunk 2 above)
Comment (indrekj) [inline templates/partials/method_summary.hamlc]: although weird that the original template also doesn't care about `.note.deprecated` class.

**Comment by indrekj** [comment]
@deiwin Updated. Changed the property style to match the regular function style instead the other way around (see updated picture in the PR description)

File: templates/class.hamlc [hunk 3]
```diff
@@ -91,8 +91,14 @@
                     %span.readonly.note.title Readonly
                   - unless property.getter
                     %span.writeonly.note.title Writeonly
+                  - if property.documentation.deprecated?
+                    %span.deprecated.note.title Deprecated
                 %dd.desc
-                  != @activate property.documentation.comment, @path
```
Comment (deiwin) [inline templates/class.hamlc]: You've removed the `comment`. Does the property description still get included properly?

File: templates/class.hamlc [hunk 4]
```diff
@@ -91,8 +91,14 @@
...
+                  - if property.documentation?.deprecated?
+                    %strong Deprecated.
+                    != @activate property.documentation.deprecated, @path, true
+                  - else if property.documentation?.summary
```
Comment (deiwin) [inline templates/class.hamlc]: Why is this an `else if`?

File: templates/class.hamlc (hunk 3 above)
Comment (indrekj) [inline templates/class.hamlc]: Yes it does.

File: templates/class.hamlc (hunk 4 above)
Comment (indrekj) [inline templates/class.hamlc]: Used exactly the same style as with the function docs: https://github.com/salemove/codo/blob/master/themes/default/templates/partials/method_summary.hamlc#L24-L29

Comment (indrekj) [inline templates/class.hamlc]: This is to hide the summary if the function/property is deprecated

Comment (indrekj) [inline templates/class.hamlc]: hmm, this makes sense for a function because you can still open the function and see the docs. property doesn't have a separate section. I can enable it here. Do you think it should be enabled for functions as well?

**Comment by ostap0207** [comment]
Deprecated 2 times now?

**Comment by indrekj** [comment]
@ostap0207 same as with functions, see: http://js-sdk-docs.salemove.com/class/Salemove.html `setupContactOperatorButton(args)`

File: templates/class.hamlc (hunk 3 above)
Comment (deiwin) [inline templates/class.hamlc]: This didn't have `true` before at the end. Pretty sure this breaks full MD functionality.

**Comment by deiwin** [comment]
+1

**Comment by ostap0207** [comment]
Would be good to have some test page with all the options to test it straight away and attach screenshot.

**Comment by ostap0207** [comment]
!squash
